lib/_http_incoming.js

```diff
@@ -137,130 +137,101 @@ function _addHeaderLines(headers, n) {
+        return 'authorization';
+      if (field === 'Last-Modified' || field === 'last-modified')
+        return 'last-modified';
+      if (field === 'Cache-Control' || field === 'cache-control')
         return '\u0000cache-control';
-      case 'Vary':
-      case 'vary':
-        return '\u0000vary';
-      case 'Content-Encoding':
-      case 'content-encoding':
-        return '\u0000content-encoding';
-      case 'Origin':
-      case 'origin':
-        return '\u0000origin';
-      case 'Upgrade':
-      case 'upgrade':
-        return '\u0000upgrade';
-      case 'Expect':
-      case 'expect':
-        return '\u0000expect';
-      case 'If-Match':
-      case 'if-match':
-        return '\u0000if-match';
-      case 'If-None-Match':
-      case 'if-none-match':
+      if (field === 'If-None-Match' || field === 'if-none-match')
         return '\u0000if-none-match';
-      case 'Accept':
-      case 'accept':
-        return '\u0000accept';
-      case 'Accept-Encoding':
-      case 'accept-encoding':
+      break;
+    case 14:
+      if (field === 'Content-Length' || field === 'content-length')
+        return 'content-length';
+      break;
+    case 15:
+      if (field === 'Accept-Encoding' || field === 'accept-encoding')
         return '\u0000accept-encoding';
-      case 'Accept-Language':
-      case 'accept-language':
+      if (field === 'Accept-Language' || field === 'accept-language')
         return '\u0000accept-language';
-      case 'X-Forwarded-For':
-      case 'x-forwarded-for':
+      if (field === 'X-Forwarded-For' || field === 'x-forwarded-for')
         return '\u0000x-forwarded-for';
-      case 'X-Forwarded-Host':
-      case 'x-forwarded-host':
+      break;
+    case 16:
+      if (field === 'Content-Encoding' || field === 'content-encoding')
+        return '\u0000content-encoding';
+      if (field === 'X-Forwarded-Host' || field === 'x-forwarded-host')
         return '\u0000x-forwarded-host';
-      case 'X-Forwarded-Proto':
-      case 'x-forwarded-proto':
+      break;
+    case 17:
+      if (field === 'If-Modified-Since' || field === 'if-modified-since')
+        return 'if-modified-since';
+      if (field === 'Transfer-Encoding' || field === 'transfer-encoding')
+        return '\u0000transfer-encoding';
+      if (field === 'X-Forwarded-Proto' || field === 'x-forwarded-proto')
         return '\u0000x-forwarded-proto';
-      default:
-        if (low)
-          return '\u0000' + field;
-        field = field.toLowerCase();
-        low = true;
-    }
+      break;
+    case 19:
+      if (field === 'Proxy-Authorization' || field === 'proxy-authorization')
+        return 'proxy-authorization';
+      if (field === 'If-Unmodified-Since' || field === 'if-unmodified-since')
+        return 'if-unmodified-since';
+      break;
+  }
+  if (lowercased) {
+    return '\u0000' + field;
+  } else {
+    return matchKnownFields(field.toLowerCase(), true);
```

Comment:
It might be slightly faster to use iteration than recursion. it might also be a wash, however. :-)

Reply:
I've run the benchmark of iteration and found there was no difference between them.